uaclient/apt.py

```diff
@@ -757,17 +766,20 @@ def is_installed(pkg: str) -> bool:
     return pkg in get_installed_packages_names()
 
 
+def get_installed_packages_gen() -> Generator[InstalledAptPackage, None, None]:
+    with PreserveAptCfg(get_apt_pkg_cache) as cache:
+        for package in cache.packages:
+            installed_version = package.current_ver
+            if installed_version:
+                yield InstalledAptPackage(
```

Comment:
I was only trying to keep the scope small since I was squeezing in this fix. If we change the return type we need to change all the callers - we should do that but I didn't have time.

Reply:
No problem we can refactor this later - good to go imho